src/parser.c

```diff
@@ -17409,16 +17409,13 @@ static bool ts_lex(TSLexer *lexer, TSStateId state) {
           ('a' <= lookahead && lookahead <= 'f')) ADVANCE(89);
       END_STATE();
     case 19:
-      if (lookahead != 0 &&
-          lookahead != '\n') ADVANCE(3);
+      if (lookahead != 0) ADVANCE(3);
       END_STATE();
     case 20:
-      if (lookahead != 0 &&
-          lookahead != '\n') ADVANCE(14);
+      if (lookahead != 0) ADVANCE(14);
       END_STATE();
     case 21:
-      if (lookahead != 0 &&
-          lookahead != '\n') ADVANCE(17);
+      if (lookahead != 0) ADVANCE(17);
```

Comment:
This parser diff gives me confidence that this was the right way to handle this

Reply:
Woah -- does this mean treesitter is parsing the JS regular expressions and re-compiling them into plain old C code? That's pretty rad.

Reply:
That's exactly what it does! It's neat until they don't support something you wanted to use 😢